dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/index.ts

```diff
@@ -1,36 +1,61 @@
 import { HttpStatusCode } from 'axios';
 import { Response } from 'express';
+import { FindOptionsWhere, Like } from 'typeorm';
+import { UtilisationReportEntity } from '@ukef/dtfs2-common';
 import { UtilisationReportRepo } from '../../../../repositories/utilisation-reports-repo';
 import { UtilisationReportReconciliationDetails } from '../../../../types/utilisation-reports';
 import { CustomExpressRequest } from '../../../../types/custom-express-request';
 import { NotFoundError, ApiError } from '../../../../errors';
 import { mapUtilisationReportEntityToReconciliationDetails } from './helpers';
+import { REGEX } from '../../../../constants';
 
 export type GetUtilisationReportReconciliationDetailsByIdRequest = CustomExpressRequest<{
   params: {
     reportId: string;
   };
+  query: {
+    facilityIdQuery?: string;
+  };
 }>;
 
 type ResponseBody = UtilisationReportReconciliationDetails | string;
 
 export const getUtilisationReportReconciliationDetailsById = async (req: GetUtilisationReportReconciliationDetailsByIdRequest, res: Response<ResponseBody>) => {
   const { reportId } = req.params;
+  const { facilityIdQuery } = req.query;
 
   try {
+    const feeRecordFindOptions: FindOptionsWhere<UtilisationReportEntity> =
+      facilityIdQuery && REGEX.UKEF_PARTIAL_FACILITY_ID_REGEX.test(facilityIdQuery)
+        ? {
+            feeRecords: {
+              facilityId: Like(`%${facilityIdQuery}%`),
```

Comment:
Now that this is even more complicated I think it warrants a custom repo method so the controller itself doesn't need to know so much about the db interaction